e2etest/bgptests/bgp.go

```diff
@@ -1136,6 +1136,35 @@ var _ = ginkgo.Describe("BGP", func() {
 					),
 				)
 			}
+
+			ginkgo.By("validating speakers running-config is updated after deleting the BFD profile")
+			configData.BFDProfiles = nil
+			for i := range configData.Peers {
+				configData.Peers[i].BFDProfile = ""
+			}
+
+			err = ConfigUpdater.Update(configData)
+			framework.ExpectNoError(err)
+
+			for _, pod := range speakerPods {
+				podExecutor := executor.ForPod(pod.Namespace, pod.Name, "frr")
+
+				Eventually(func() string {
+					// We need to assert against the output of the command as a bare string, as
+					// there is no json version of the command.
+					cfgStr, err := podExecutor.Exec("vtysh", "-c", "show running-config")
+					if err != nil {
+						return err.Error()
+					}
+
+					return cfgStr
+				}, 1*time.Minute).Should(
+					And(
+						ContainSubstring("log file /etc/frr/frr.log informational"),
+						WithTransform(substringCount("\n profile fullbfdprofile1"), Equal(0)),
```

Comment:
Can we also validate that the BGP peer does not have BFD enabled with the default BFD profile?

This is, should not have `neighbor $NEIGHBOR_IP bfd`

Reply:
I added the test.
It seems we found a bug in FRR. The BFD profile is removed from the FRR config but the BGP peers still have BFD enabled with: `neighbor $NEIGHBOR_IP bfd`.

Reply:
Can you check that the renderer file does not contain them (i.e. it's an frr bug and not a bug on our side)?

Reply:
I checked the rendered file, it doesn't have BFD configuration at all.
We tested it manually as well and got the same result.